atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -422,28 +456,32 @@ public void testPutCleanup() throws Exception {
             inSequence(seq);
         }});
 
-        Transaction snapshot = transactionWrapper.apply(new SnapshotTransaction(metricsManager,
-                keyValueService,
-                null,
-                transactionService,
-                NoOpCleaner.INSTANCE,
-                () -> transactionTs,
-                ConflictDetectionManagers.create(keyValueService),
-                SweepStrategyManagers.createDefault(keyValueService),
-                transactionTs,
-                Optional.empty(),
-                PreCommitConditions.NO_OP,
-                AtlasDbConstraintCheckingMode.NO_CONSTRAINT_CHECKING,
-                null,
-                TransactionReadSentinelBehavior.THROW_EXCEPTION,
-                false,
-                timestampCache,
-                getRangesExecutor,
-                defaultGetRangesConcurrency,
-                MultiTableSweepQueueWriter.NO_OP,
-                MoreExecutors.newDirectExecutorService(),
-                true,
-                () -> transactionConfig));
+        AtomicInteger counter = new AtomicInteger();
```

Comment:
from what I can see, you don't really do anything with the counter besides pass it in, is it not sufficient to just include it in the delegate?

Reply:
In general it has to be shared, even now with an interface.